aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -235,6 +236,25 @@ Tensor& cumprod_out(Tensor& result, const Tensor& self, int64_t dim, c10::option
   return result;
 }
 
+// Implement std::nan<IntegralType> for MSVC according to https://zh.cppreference.com/w/cpp/numeric/math/isnan.
+namespace {
+#ifdef _MSC_VER
+template<typename T>
+typename std::enable_if<std::is_integral<T>::value, bool>::type isnan_(T x) {
+  return std::isnan(static_cast<double>(x));
```

Comment:
I see THCNumerics define casts all input to float for all dtypes https://github.com/pytorch/pytorch/blob/d13c1b8af803eecb7586c3760e6e93efb51e6677/aten/src/THC/THCNumerics.cuh#L304 should we follow that here as well?

Reply:
looks good to me otherwise! thanks for the fix :)

Reply:
Okay, let's follow suit.

Reply:
Well, I check that code carefully. The quoted code is the implementation for the type `at::Half`. Regarding `isnan` for integral types, it always returns `false`. Since `at::Half` is not involved here, I think it should be okay.